proxy/src/main/java/com/velocitypowered/proxy/protocol/netty/GameSpyQueryHandler.java

```diff
@@ -212,12 +210,9 @@ private List<QueryResponse.PluginInformation> getRealPluginInformation() {
     return result;
   }
 
-  private static class ResponseWriter {
-
-    private final ByteBuf buf;
-    private final boolean isBasic;
+  private record ResponseWriter(ByteBuf buf, boolean isBasic) {
 
-    ResponseWriter(ByteBuf buf, boolean isBasic) {
+    private ResponseWriter(ByteBuf buf, boolean isBasic) {
```

Comment:
To do validations on a constructor of a record, you can simply do this

![image](https://github.com/PaperMC/Velocity/assets/68704415/0f4bb826-b735-405c-a074-e91da44492d9)